lib/vector-core/src/schema/requirement.rs

```diff
@@ -72,11 +72,17 @@ impl Requirement {
     }
 
     /// Validate the provided [`Definition`] against the current requirement.
+    /// If `validate_schema_type` is true, validation ensure the types match,
+    /// otherwise it only ensures the required fields exist.
     ///
     /// # Errors
     ///
     /// Returns a list of errors if validation fails.
-    pub fn validate(&self, definition: &Definition) -> Result<(), ValidationErrors> {
+    pub fn validate(
```

Comment:
As we discussed yesterday, I believe `definition.invalid_meaning(identifier).cloned()` will return false positives for situations where you can have multiple different paths for a single meaning. I'm fine if the fix isn't included in this PR, but it should be added in a follow up.